lib/libc/picolibc/CMakeLists.txt

```diff
@@ -22,8 +22,13 @@ if(NOT CONFIG_PICOLIBC_USE_MODULE)
   # Use picolibc provided with the toolchain. This requires a new enough
   # toolchain so that the version of picolibc supports auto-detecting a
   # Zephyr build (via the __ZEPHYR__ macro) to expose the Zephyr C API
-  zephyr_compile_options(PROPERTY specs picolibc.specs)
-  zephyr_link_libraries(PROPERTY specs picolibc.specs)
+  execute_process(
+    COMMAND ${CMAKE_C_COMPILER} -print-file-name=picolibc.specs
+    OUTPUT_VARIABLE PICOLIBC_SPECS
+    OUTPUT_STRIP_TRAILING_WHITESPACE
+  )
+  zephyr_compile_options(PROPERTY specs "${PICOLIBC_SPECS}")
+  zephyr_link_libraries(PROPERTY specs "${PICOLIBC_SPECS}")
```

Comment:
This now invokes the compiler command with `-print-file-name=picolibc.specs`, which may or may not work on the toolchains other than GCC; at least, Clang just returns the specified file name for non-existent files; but, there is no guarantee that other compilers will do the same.

Even for Clang, although this does not explicitly fail, this entire section should be guarded by `if(CMAKE_C_COMPILER_ID STREQUAL "GNU")` since it is not applicable to Clang.